Add a Discovery Service section

ptgott opened the pull request:
Closes #38741

Add a dedicated sidebar section for the Teleport Discovery Service. Since the Discovery Service is a separate agent service from the SSH Service, Kubernetes Service, etc., it makes sense to add a dedicated section for this service. In the future, we can include configuration guides, architecture guides, etc. in this section.

To create the new section, this change moves existing auto-discovery guides from other sections of the docs.

ptgott committed:
Add a Discovery Service section

Closes #38741

Add a dedicated sidebar section for the Teleport Discovery Service.
Since the Discovery Service is a separate agent service from the SSH
Service, Kubernetes Service, etc., it makes sense to add a dedicated
section for this service. In the future, we can include configuration
guides, architecture guides, etc. in this section.

To create the new section, this change moves existing auto-discovery
guides from other sections of the docs.

ptgott commented:
In the spirit of starting simple and iterating, this PR doesn't do much else than move existing pages and create bare-bones menu pages, but we can open further issues to make more specific changes like adding a configuration guide.

hugoShaka commented on docs/pages/auto-discovery/introduction.mdx:
```diff
@@ -0,0 +1,22 @@
+
+Set up Teleport auto-discovery for resources in your infrastructure:
+
+- [Servers](./servers.mdx)
+- [Kubernetes clusters](./kubernetes.mdx)
+- [Databases](./databases.mdx)
```
I think the Kubernetes application discovery guide should also be moved there: https://goteleport.com/docs/application-access/enroll-kubernetes-applications/get-started/

ptgott committed:
Respond to hugoShaka feedback

- Move the Kubernetes application discovery docs
- Edit the "Kubernetes Access" intro to include a link to Kubernetes
  auto-discovery
